api/v1/controllers/AuthController.js

```diff
@@ -70,8 +80,85 @@ function refreshToken (req, res, next) {
 		});
 }
 
+function requestPasswordReset (req, res, next) {
+	var userEmail = req.query.email;
+
+	UserService
+		.findUserByEmail(userEmail)
+		.then(function (user){
+			return AuthService.generateToken(user, scopes.AUTH);
+		})
+		.then(function (tokenVal){
+			//TODO: Determine exact url route for reset page
+			tokenURL = (config.isDevelopment ? "localhost:8080/" : "hackillinois.org/") + "v1/auth/resetpage?token=" + tokenVal;
+			var substitutions = {'resetURL': tokenURL};
+			return MailService.send(userEmail, mail.templates.passwordReset, substitutions);
+		})
+		.then(function(){
+			res.body = {};
+			next();
+			return null;
+		})
+		.catch(function (error){
+			next(error);
+			return null;
+		});
+}
+
+function passwordReset(req, res, next) {
+	// MARK - Should be able to remove this and let middleware handle the requirements
+	/* 
+	if (req.body.password.length < MINIMUM_PASSWORD_LENGTH)
+	{
+		var message = 'Password must be at least ' + MINIMUM_PASSWORD_LENGTH + ' characters long';
+		next(new errors.InvalidParameterError(message, 'password'));
+		return;
+	}
+	*/
+
+	AuthService
+		.findTokenByValue(req.body.token)
```

Comment:
Most of this looks like business logic, so I'll request that this and everything below is moved to a `resetPassword` method in the UserService (which takes a user and the new password).